app.rb

```diff
@@ -43,52 +48,57 @@ def add_book
     puts 'Author'
     author = Input.new.input
     book = Book.new(title, author)
+    @books = []
     @books.push(book)
+    write('Data Files/books.json', @books)
     puts 'Book added successfully.'
   end
 
   def add_rental
     puts 'Select a book from the following list by number'
+    @books = read('Data Files/books.json')
     @books.each_with_index do |book, index|
-      puts "#{index}) Title: #{book.title}, Author: #{book.author}"
+      puts "#{index}) Title: #{book['title']}, Author: #{book['author']}"
     end
     book_index = Input.new.input_int
     puts 'Select a person from the following list by number (not Id)'
+    @persons = read('Data Files/people.json')
     @persons.each_with_index do |person, index|
-      puts "#{index}) Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      puts "#{index}) Name: #{person['name']}, ID: #{person['id']}, Age: #{person['age']}"
     end
     person_index = Input.new.input_int
     puts 'Date'
     date = Input.new.input
     rental = Rental.new(date, @books[book_index], @persons[person_index])
+    @rentals = []
     @rentals.push(rental)
+    write('Data Files/rentals.json', @rentals)
     puts 'Rental added sucessfully'
   end
 
   def list_all_people
+    @persons = read('Data Files/people.json')
```

Comment:
Instead of reading the file when a user displays the list, it would be better if you initialized `@persons`, `@books` and `@rentals` in the `initilize` method like this: 
``` 
def initialize
    @books = read('Data Files/books.json')
    @people = read('Data Files/people.json')
    @rentals = read('Data Files/rentals.json')
 end
```